lambda_backend/user_microservice/src/common/identityProvider.ts

```diff
@@ -16,16 +44,29 @@ export const adminGetCognitoUserDetails = async (Username: string): Promise<Cogn
     const UserAttributes = adminGetUserResponse.UserAttributes as AttributeType[];
     return parseUserAttributes(UserAttributes);
   } catch (error) {
-    throw createError(500, 'Error retrieving Cognito user details');
+    throw createError(500, 'Error retrieving Cognito user details', error);
+  }
+};
+
+export const getCognitoUserDetails = async (AccessToken: string): Promise<CognitoUserDetails> => {
+  const getUserParams: GetUserRequest = { AccessToken };
+  try {
+    const { UserAttributes } = await cognitoIdentity.getUser(getUserParams).promise();
+    return parseUserAttributes(UserAttributes);
+  } catch (error) {
+    throw createError(500, 'Error retrieving Cognito user details', error);
   }
 };
 
 const parseUserAttributes = (UserAttributes: AttributeType[]): CognitoUserDetails => {
   const userEmail = UserAttributes.filter((attribute) => attribute.Name === 'email')[0]
     .Value as string;
-  return { userEmail };
+  const userRole = UserAttributes.filter((attribute) => attribute.Name === 'custom:role')[0]
+    .Value as 'admin' | 'user';
+  return { userEmail, userRole };
 };
 
 interface CognitoUserDetails {
   userEmail: string;
+  userRole: 'admin' | 'user';
```

Comment:
Maybe should have an extra type like `UserRole`, we might want to add gymowner or smth next time